modules/testing/include/hephaestus/testing/random_container.h

```diff
@@ -0,0 +1,45 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <random>
+#include <vector>
+
+#include <magic_enum.hpp>
+
+#include "hephaestus/testing/random_type.h"
+
+namespace heph::testing {
+
+//=================================================================================================
+// Random vector generation
+//=================================================================================================
+namespace internal {
+template <typename T>
+concept IsVectorT =
```

Comment:
nit. this feels like a quite generic concept, maybe we can move it to a general location

Reply:
Talked this out in person. We'll leave generalizing this concept to the time when there is a proper traits module.